core/src/main/java/amino/run/kernel/client/KernelClient.java

```diff
@@ -267,13 +268,10 @@ public void measureServerMetrics() {
                 /* Measure data transfer rate by sending some arbitrary data */
                 server.receiveHeartBeat(randomBytes);
                 long t3 = System.nanoTime();
-                metric.latency = (t2 - t1) / 2;
+                metric.latency = (t2 - t1); // RTT
```

Comment:
Latency needs to be 1-way latency, so the original code was correct.

Reply:
Latency is not implicitly equal to half of RTT, because delay may be asymmetrical between two given endpoints.  As we make RPC calls to destination and then wait for an acknowledgement/result to come back before making further calls, RTT may be better  ?
Was searching about it and found few links treating that way -
https://www.sas.co.uk/blog/what-is-network-latency-how-do-you-use-a-latency-calculator-to-calculate-throughput
Let me know your opinion